cache_flat_mutation_reader.hh

```diff
@@ -510,7 +510,7 @@ bool cache_flat_mutation_reader::ensure_population_lower_bound() {
             return false;
         }
 
-        if (cmp(cur.position(), _last_row.position()) != 0) {
+        if (cmp(cur.table_position(), _last_row.position()) != 0) {
             return false;
```

Comment:
This one looks as if a simple unit test can be constructed for it (I'm not saying that creating the unit test would be simple, just that in the end it would probably be a large sequence of range tombstones or something).